module/support/optimisation/NSGA2Optimiser/src/nsga2/Population.cpp

```diff
@@ -0,0 +1,222 @@
+#include "Population.hpp"
+
+#include <nuclear>
+
+namespace nsga2 {
+    Population::Population(const int& _size,
```

Comment:
What's the purpose of prefixing all of these with underscores? Pls also fix casing :)

Reply:
The underscore is a naming convention. Which has the meaning of these objects are used internally. Eg size = _size

Reply:
Generally you should put these underscores at the end of the variable rather than the start. "technically" variables with one or two underscores are reserved by the compiler and standard library or something.

So generally it's safer to not prepend underscores.

https://releases.llvm.org/11.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/bugprone-reserved-identifier.html

Reply:
Ah ok. I was taught to put them at the front when I first started learning C++ back in 1998. Thanks Trent for the reference.

Reply:
(Insanely pedantic "ackshually" comment warning)

_ at the start followed by a capital letter and double underscores in any part of the identifier are all reserved for the implementation, so making user defined ones is technically undefined behaviour, iirc.

So I think this should technically be ok, since it's not _[A-Z], but definitely agree with Trent that at the end is the way to go

Reply:
All good. Thanks Kip. I've learnt something new

Reply:
Either way, it's getting into pedantic levels, you're almost guaranteed not to collide with something in use by the standard. But it's easier to just avoid the whole issue by putting them on the end.

For an example of the crazy behaviour see `pure_evil` where we use that power for "good"

Reply:
I have looked at that file before. How could I not when it's called pure_evil. I'll fix them up and learn by it. Thanks Trent